### DISCUSSION TITLE
Problems setting auto=route on a Linux client

### DISCUSSION THREAD

**Discussion by acrolink** [strongswan/strongswan#630] [Q&A]
I have this connection setting to establish a connection between a Linux client (A) and a remote VPN server (B) (here Linux client side):

```
conn ipsec-ikev2-vpn-client
    auto=start
    right=ikev2.my.server
    rightid=ikev2.my.server
    rightsubnet=0.0.0.0/0
    rightauth=pubkey
    leftsourceip=%config
    leftid=eap_10_50_10_254
    leftauth=eap-mschapv2
    eap_identity=%identity
```

All working fine but if I change `auto` to `route` then all connectivity on the Linux client is lost. I want in case B goes down for sometime, I want A to reconnect automatically. I understand that this can be accomplished by setting `auto` to `route` but then I face the issue described above. What can be wrong here?

Here's partial setup on the server side (B):

```
    children {
        net {
            inactivity = 86400s
            local_ts="10.50.10.0/24"
            esp_proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-modp3072,aes192-sha256-ecp256-modp3072,default
            close_action=start
            start_action=trap
            dpd_action = clear
        }
    }
```

**Comment by tobiasbrunner** [comment]
With `rightsubnet=0.0.0.0/0`, no traffic outside an IPsec tunnel is allowed, that is, all packets will generate acquires to trigger SAs. However, in your case, the server will narrow the traffic selector down to `10.50.10.0/24`, so only traffic to that subnet will actually get tunneled correctly. Yet that has no impact on the installed trap policy, which is still for `0.0.0.0/0` but uses the same reqid, so all traffic will be sent through the installed IPsec SA, but then gets dropped by the peer because it won't match the negotiated IPsec policy on its end. The easiest fix is to configure `rightsubnet=10.50.10.0/24` on the client to avoid the narrowing.

     close_action=start
     start_action=trap

Neither does really make sense on the server, unless it knows the IP address of the client so it could connect to it.

**Reply by acrolink** [reply]
Seems that this is working on the client side, now (connection re-established):

```
conn ipsec-ikev2-vpn-client
    auto=route
    closeaction=restart
    right=ikev2.my.server
    rightid=ikev2.my.server
    rightsubnet=10.50.10.0/24
    rightauth=pubkey
    leftsourceip=%config
    leftid=eap_10_50_10_254
    leftauth=eap-mschapv2
    eap_identity=%identity
```
But the problem is that after I do a `systemctl restart strongswan-starter` I have to do also a `ipsec up ipsec-ikev2-vpn-client` to bring the connection up. Does not start automatically for some reason.

**Reply by tobiasbrunner** [reply]
It's triggered by _traffic_ to the remote subnet.

**Reply by acrolink** [reply]
> It's triggered by _traffic_ to the remote subnet.

I see. What I have done now to make it work the way I expect is to ping an address on the remote subnet every minute (using cron). This way it remains connected all the time.
